### PR TITLE
Add support for nested profiles

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Adds support for nested profiles:

```
    - name: nested-profile
      kind: Profile
      profile:
        url: https://github.com/weaveworks/nginx-profile
        branch: helm-artifact
```

### Checklist
- [x] Added tests that cover your change
- [ ] Added/modified documentation as required (such as the `README.md`)
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] Added a `kind` label to the PR (e.g. `kind/feature`)

### CODE REVIEW

File: examples/profile-subscription.yaml [hunk 1]
```diff
@@ -5,4 +5,4 @@ metadata:
   namespace: default
 spec:
   profileURL: https://github.com/weaveworks/nginx-profile
-  branch: main
+  branch: move-helm-to-nested
```
Comment (aclevername) [inline examples/profile-subscription.yaml:8]: TODO revert back to main when https://github.com/weaveworks/nginx-profile/pull/7 is merged

File: api/v1alpha1/profile_types.go [hunk 2]
```diff
@@ -73,6 +85,14 @@ type Chart struct {
 	Version string `json:"version,omitempty"`
 }
 
+// Chart defines properties to access remote helm charts.
```
Comment (Callisto13) [inline api/v1alpha1/profile_types.go:88]: nope

File: controllers/profile_controller_test.go [hunk 3]
```diff
@@ -23,8 +23,10 @@ var _ = Describe("ProfileController", func() {
 	const helmChartURL = "https://charts.bitnami.com/bitnami"
 
 	var (
-		namespace string
-		ctx       = context.Background()
+		namespace            string
+		branch               = "move-helm-to-nested"
```
Comment (Callisto13) [inline controllers/profile_controller_test.go:27]: is this also a TODO?

File: pkg/profile/artifact.go [hunk 4]
```diff
@@ -183,6 +183,22 @@ func (p *Profile) MakeOwnerlessArtifacts() ([]runtime.Object, error) {
...
+				return nil, fmt.Errorf("profile cannot contain profile artifact pointing to itself")
+			}
+			nestedProfileDef, err := getProfileDefinition(artifact.Profile.URL, artifact.Profile.Branch, p.log)
+			if err != nil {
+				return nil, fmt.Errorf("failed to fetch profile %q: %w", artifact.Name, err)
+			}
+			nestedProfile := p.subscription.DeepCopyObject().(*profilesv1.ProfileSubscription)
+			nestedProfile.Spec.ProfileURL = artifact.Profile.URL
+			nestedProfile.Spec.Branch = artifact.Profile.Branch
+			nestedObjs, err := New(p.ctx, nestedProfileDef, *nestedProfile, p.client, p.log).MakeOwnerlessArtifacts()
```
Comment (Callisto13) [inline pkg/profile/artifact.go:197]: so this is basically "flattening" the nested artifacts so it's like they were just listed in the "top-level" profile?

File: tests/acceptance/acceptance_test.go [hunk 5]
```diff
@@ -40,7 +40,7 @@ var _ = Describe("Acceptance", func() {
 
 		BeforeEach(func() {
 			profileURL = "https://github.com/weaveworks/nginx-profile"
-			branch = "main"
+			branch = "move-helm-to-nested"
```
Comment (Callisto13) [inline tests/acceptance/acceptance_test.go:43]: another TODO

File: pkg/profile/artifact.go (hunk 4 above)
Comment (aclevername) [reply pkg/profile/artifact.go:197]: yes

File: tests/acceptance/acceptance_test.go (hunk 5 above)
Comment (aclevername) [reply tests/acceptance/acceptance_test.go:43]: yeah its all over to get the tests passing

File: controllers/profile_controller_test.go (hunk 3 above)
Comment (aclevername) [reply controllers/profile_controller_test.go:27]: yes part of the other TODO

File: api/v1alpha1/profile_types.go (hunk 2 above)
Comment (aclevername) [reply api/v1alpha1/profile_types.go]: fixed!

**Review by Callisto13 — APPROVED** [review]
lgtm

**Pull request merged**